Notes/Algorithms/Data Structures/Hash-Tables.c

```diff
@@ -1,37 +1,345 @@
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 #include <stdbool.h>
 
 #define SUCCESS 0
 #define ERROR -1
-#define OVERFLOW -2
-#define EMPTY -3
+// Defined limit for a string key length 
+#define LIMIT 10
 
-// Hash struct
-typedef struct Hash {
-    int key;
+// Hash Item struct
+typedef struct Item {
+    char* key;
     int value;
-} Hash;
+} Item;
 
-void print() {
+// Hash Table struct
+typedef struct HashTable {
+    int size;
+    Item** items;
+} HashTable;
 
+
+// returns true if a collision would happen for the initially hashed index for a key/value pair and false if no collision would occurred 
+bool isCollision(HashTable * table, int index) {
+    if (table->items[index] != NULL) {
+        return true;
+    }
+    return false;
 }
 
-void traverse() {
+// equals checks the testHash and expectedHash for parity
+// if the two HashTables match returns true
+// if a mismatch occurs returns false
+bool equals(HashTable * testHash, HashTable * expectedHash) {
+    for (int i = 0; i < expectedHash->size; i++) {
+        if (strcmp(testHash->items[i]->key, expectedHash->items[i]->key) != 0 || testHash->items[i]->value != expectedHash->items[i]->value) {
```

Comment:
Interesting test case! Added a check for empty items[i] for segfaults. Also added a mini test case function. I made it a standalone since I did not think it really felt in any of the bigger build Test functions. [commit e1115c34634aee9f41169dc13459008361fc1844](https://github.com/kcarrel/Training-Notes/pull/12/commits/e1115c34634aee9f41169dc13459008361fc1844)